pkg/sql/catalog/systemschema/system.go

```diff
@@ -243,6 +243,77 @@ CREATE TABLE system.jobs (
 	FAMILY claim (claim_session_id, claim_instance_id, num_runs, last_run)
 );`
 
+	// JobProgressTableSchema is a table that contains one row per job reflecting
+	// that job's current progress for human consumption (e.g. in SHOW or the UI).
+	// This table includes the time the progress was reported in the PK, so to
+	// update the current progress for a job, one would DELETE the old row and
+	// INSERT a new one, rather than UPDATEing the existing row. This is done to
+	// ensure revisions of the progress are not revisions to one "row" in CRDB,
+	// and thus the ranges can split between revisions to progress even though
+	// they cannot split between revisions to a SQL row. This is the same pattern
+	// used by job_info.
+	JobProgressTableSchema = `
+CREATE TABLE system.job_progress (
+	job_id INT8 NOT NULL,
+	written TIMESTAMPTZ NOT NULL DEFAULT now(), -- when this progress was reported.
+	fraction FLOAT, -- the fraction of some known work that is complete (e.g. half done with backfill)
+	resolved DECIMAL, -- a timestamp up to which processing is complete (aka "highwater"/resolvedTS/etc). Typically used by perpetual jobs instead of fraction.
+	--
+	CONSTRAINT "primary" PRIMARY KEY (job_id ASC, written DESC),
+)`
+
+	// JobProgressHistoryTableSchema is identical to job_progress, but is allowed
+	// to contain multiple rows for the same job_id. This table is used to track
+	// the progress of a job over time, whereas job_progress only contains the
+	// most recent progress update to allow more efficient lookups of the current
+	// progress in joins for SHOW JOBS. This table generally should only queried
+	// for a specific job at a time. History is not stored indefinitely and may be
+	// pruned based on retention limits.
+	JobProgressHistoryTableSchema = `
+	CREATE TABLE system.job_progress_history (
+		job_id INT8 NOT NULL,
+		written TIMESTAMPTZ NOT NULL DEFAULT now(),
+		fraction FLOAT,
+		resolved DECIMAL,
+		--
+		CONSTRAINT "primary" PRIMARY KEY (job_id ASC, written DESC),
+	)`
+
+	// JobStatusTableSchema is the table that contains the current, user-visible
+	// status of a job, which is typically its current phase of execution such as
+	// such as "backfilling" or "waiting for GC". This table contains one row per
+	// job and updates to the "current status" field for a job replace it with a
```

Comment:
theres a typo around "replace"

Reply:
where?

I also reworded some of this anyway just to make the status vs message distinction clearer.